### DISCUSSION TITLE
Build failed with Java 17

### DISCUSSION THREAD

**Discussion by alegos27** [codemagic-ci-cd/codemagic-docs#2435] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

replace this:
```
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
```

with this:    
```
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }
```

and build on codemagic (local build run successful)

### Expected results

Successful build

### Actual results

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':connectivity_plus:compileReleaseJavaWithJavac'.
> error: invalid source release: 17

### Build id (optional)

66012bea5eb9f6c6d5367943

**Comment by dtrdic** [comment, marked as answer]
hey @alegos27 if you need to instruct Codemagic to build your app with the java 17 version, just specify `JAVA_HOME` under the environment variables section to this path `/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home` and rerun the build.

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/9720832e-9f5a-4f50-bdfa-f7f04e4340ac)

**Comment by Ultranmus** [comment]
@alegos27 what is group name for this?

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/140942235/db2fe249-11ca-4199-8101-f6679c48e30c)

**Reply by dtrdic** [reply]
@Ultranmus variable group is an option that is only available if you are using codemagic.yaml configuration, the above is a solution if you are using Workflow editor
![Screenshot 2024-03-27 at 15 51 28](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/f02b0d02-e036-40f7-8db3-516801a58646)

If you are indeed using a codemagic.yaml configuration, then you simply need to specify the Java version under the environment section in your YAML like this, and that's it
```
    android-workflow:
        name: Android Workflow
        max_build_duration: 120
        instance_type: mac_mini_m1
        environment:
          java: 17
```

**Comment by Ultranmus** [comment]
@dtrdic After setting the java version 17, now I am getting this error. 

```
== Use Flutter channel stable ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.4-stable.zip --connect-timeout 10.00 --max-time 120.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.4-stable.zip
curl: (28) Operation timed out after 120001 milliseconds with 74895448 out of 1168340724 bytes received

Downloading Flutter release timed out

Build failed :|
Downloading Flutter release timed out
```

**Comment by Ultranmus** [comment]
The java 17 issue was resolved with @dtrdic solution and the **Flutter release timed out** error was due to a current problem with the Codemagic server. #2424
